provider/datagen/src/transform/cldr/calendar/chinese_based.rs

```diff
@@ -0,0 +1,51 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use calendrical_calculations::chinese_based::{Chinese, ChineseBased, Dangi};
+use icu_calendar::provider::chinese_based::*;
+use icu_provider::datagen::IterableDataProvider;
+use icu_provider::prelude::*;
+
+const YEARS: i32 = 200;
+const ISO_START: i32 = 1900;
```

Comment:
I just made these numbers up, open to other suggestions. Data size is currently 600B each.

Previously when Andrew and I were designing it we were including the related ISO year which would have limited the range to around 200 but now this range can be arbitrarily extended to whatever we want.

Reply:
If you make it 250 years then we're right in the middle of the range.

Reply:
No, there's no cycle.

Reply:
I bumped it to 250. Also open to making it 200 and 1950, but I think prior dates are actually somewhat important especially when it comes to peoples' birthdays. Not that important though.

Reply:
Well there has to be a cycle *at some point*

Reply:
Living people's birthdays is also my rule of thumb.